apps/dashboard/services.py

```diff
@@ -326,38 +333,56 @@ def get_channel_breakdown_data(self, **filters) -> dict[str, Any]:
 
         querysets = self.get_filtered_queryset_base(**filters)
 
-        # Get platform statistics by aggregating all channels per platform
-
-        from apps.channels.models import ChannelPlatform
-
-        platform_data = []
-
-        # Get all platforms that have channels in this team
         platforms_in_use = (
             ExperimentChannel.objects.filter(team=self.team, deleted=False)
             .order_by()
             .values_list("platform", flat=True)
             .distinct()
         )
-        for platform in platforms_in_use:
-            platform_sessions = querysets["sessions"].filter(experiment_channel__platform=platform)
-            platform_messages = querysets["messages"].filter(
-                chat__experiment_session__experiment_channel__platform=platform
+        sessions_stats = (
+            querysets["sessions"]
+            .values("experiment_channel__platform")
+            .annotate(sessions_count=Count("id"), participants_count=Count("participant", distinct=True))
+        )
```

Comment:
Because of how the `querysets["sessions"]` is being created this actually produces results grouped by date which isn't what you want here. One trick you can do is use `order_by()` to clear the distinct fields and then add `distinct=True` to the count:

```suggestion
        sessions_stats = (
            querysets["sessions"].order_by()
            .values("experiment_channel__platform")
            .annotate(sessions_count=Count("id", distinct=True), participants_count=Count("participant", distinct=True))
        )
```